### ISSUE TITLE
[Feature Request] Add support for boolean parameters in custom strategies.

### ISSUE THREAD

**Issue by GrgDev** [Unleash/unleash#435]
Right now using v3.1.1 if I want to add a new parameter to a custom strategy, my choices for parameter types are string, number, list, and percentage. I have a use case where a user has a series of boolean values associated with them and I would like to make a custom strategy that turns on a feature if they have a certain combination subset of those boolean values, otherwise the feature remains off.

The problem right now is that there is no explicit boolean value supported as a strategy parameter even though that booleans are generally considered to be a basic and standard data type in most contexts. There are ways for me to personally work around this. So far workaround ideas include:

- Using percentages instead of booleans and any non-zero value is treated as True
- Using strings and have them set to "true" if True, otherwise false for any other value
- Using a number where the value is a boolean bitmask, similar to octal file permissions

While I am willing to live with these workarounds, those that I am trying to get to use the Unleash app for their stuff complain about how they have to set parameters in such a roundabout way just to get a boolean value to work with.

I don't see where in the source the built-in strategy parameter types are defined, but if someone could point me in that direction as well so I can see how difficult it would be to add a boolean type, I would appreciate it.

**Comment by ivarconr** [comment]
Adding support for boolean value types sounds like a great idea, I will look in to tomorrow, believe this should be pretty easy to add in the UI.

**Comment by ivarconr** [comment]
Ok @GrgDev I just had to do a test on this. Does this look like something you can work with:

![image](https://user-images.githubusercontent.com/158948/57323353-eeff6180-7105-11e9-94e6-0dc54af94f5b.png)


The client API would have the following structure. 

```json
{
   "name": "Demo",
   "description": "hello world",
   "enabled": false,
   "strategies": [
      {
         "name": "Boolean",
         "parameters": {
            "isBeta": "true",
            "isOld": "false",
            "percentage": 50
         }
      }
   ],
   "variants": null,
   "createdAt": "2019-04-30T20:34:31.690Z"
}
```

The protocol is kind of locket to string for parameter type, so you will still have to parse "true" / "false" in the client. I don't want to break this at this time as I am working on a more flexible kind of strategies. 

Would solition solve it for you?

**Comment by GrgDev** [comment]
Yeah, that would be a great improvement. I might still have to parse the strings but at least I'd be guaranteed an exact two possibilities of string values instead of a possible human error making a typo and putting in "fals" instead.

**Comment by ivarconr** [comment]
You can now play with the feature here. Will release soon;
http://unleash.herokuapp.com/#/features/strategies/Demo

**Comment by ivarconr** [comment]
Released as part of unleash-server v3.2.10
